e2e/nomostest/testpredicates/predicates.go

```diff
@@ -894,3 +887,61 @@ func WatchSyncPredicate() (Predicate, <-chan struct{}) {
 		return nil
 	}, syncCh
 }
+
+// DeploymentContainerImageEquals returns a predicate that errors if the
+// deployment does not have a container with the specified name and image.
+func DeploymentContainerImageEquals(containerName, image string) Predicate {
+	return func(o client.Object) error {
+		if o == nil {
+			return ErrObjectNotFound
+		}
+		d, ok := o.(*appsv1.Deployment)
+		if !ok {
+			return WrongTypeErr(d, &appsv1.Deployment{})
+		}
+		reconcilerContainer := ContainerByName(d, containerName)
+		if reconcilerContainer == nil {
+			return fmt.Errorf("expected container not found: %s", containerName)
+		}
+		if reconcilerContainer.Image != image {
+			return fmt.Errorf("expected %q container image to equal: %s, got: %s",
+				containerName, image, reconcilerContainer.Image)
+		}
+		return nil
+	}
+}
+
+// DeploymentContainerPullPolicyEquals returns a predicate that errors if the
+// deployment does not have a container with the specified name and
+// imagePullPolicy.
+func DeploymentContainerPullPolicyEquals(containerName string, policy corev1.PullPolicy) Predicate {
+	return func(o client.Object) error {
+		if o == nil {
+			return ErrObjectNotFound
+		}
+		d, ok := o.(*appsv1.Deployment)
+		if !ok {
+			return WrongTypeErr(d, &appsv1.Deployment{})
+		}
+		reconcilerContainer := ContainerByName(d, containerName)
```

Comment:
nit: this is a generic function, variable name should be generic. e.g. `container` instead of `reconcilerContainer`

ditto above

Reply:
fixed